src/hotspot/share/utilities/istream.cpp

```diff
@@ -0,0 +1,378 @@
+    set_done();
+    _input_state = IState::ERR_STATE;
+    assert(error(), "");
+  } else if (error()) {
+    _input_state = definitely_done() ? IState::EOF_STATE : IState::NTR_STATE;
+  }
+}
+
+void inputStream::clear_buffer() {
+  _content_end = _beg = _end = _next = 0;
+  _line_ending = 0;
+}
+
+const char* inputStream::next_content(size_t& next_content_length) const {
+  assert(is_sane(), "");
+  size_t len = buffered_content_length(false);
+  next_content_length = len;
+  return len == 0 ? "" : &_buffer[_next];
+}
+
+void inputStream::set_input(inputStream::Input* input) {
+  clear_buffer();
+  if (_input != nullptr && _input != input) {
+    _input->close();
+  }
+  _input = input;
+  _input_state = IState::NTR_STATE;
+}
+
+bool inputStream::fill_buffer() {
+  assert(!definitely_done(), "");  // caller responsibility
+  while (need_to_read()) {
+    size_t fill_offset, fill_length;
+    prepare_to_fill_buffer(fill_offset, fill_length);
+    if (error())  return false;
+    assert(fill_length > 0, "");
+    assert(fill_offset < _buffer_size, "");
+    assert(fill_offset + fill_length <= _buffer_size, "");
+    size_t nr = 0;
+    if (_input != nullptr && _input_state == IState::NTR_STATE) {
+      nr = _input->read(&_buffer[fill_offset], fill_length);
+      if (nr == 0)  _input_state = IState::EOF_STATE;  // do not get EOF twice
+    }
+    bool last_partial = false;
+    if (nr > 0) {
+      fill_offset += nr;
+    } else if (_beg == _end) {  // no partial line, so end it now
+      // we hit the end of the file (or there was never anything there)
+      COV(FIB_P);
+      assert(!definitely_done(), "");
+      set_done();
+      assert(definitely_done(), "");
+      return false;
+    } else {
+      // pretend to read a newline, to complete the last partial line
+      COV(FIB_E);
+      _buffer[fill_offset++] = '\n';  // insert phantom newline
+      last_partial = true;
+    }
+    set_buffer_content(_beg, fill_offset);
+    assert(!definitely_done(), "");
+    if (need_to_read()) { COV(FIB_N); }
+    else                { COV(FIB_L); }
+    if (last_partial) {
+      assert(have_current_line(), "");
+      _line_ending = 0;
+      _content_end -= 1;  // reverse insertion of phantom newline
+      assert(_next == _content_end + NEXT_PHANTOM, "");
+      assert(have_current_line(), "");
+    }
+  }
+  return true;
+}
+
+// Find some space in the buffer for reading.  If there is already a
+// partial line in the buffer, new space must follow it immediately.
+// The partial line is between _beg and _end, and no other parts of
+// the buffer are in use.
+void inputStream::prepare_to_fill_buffer(size_t& fill_offset,
+                                         size_t& fill_length) {
+  assert(need_to_read(), "");  // _next pointer out of the way
+  if (_buffer_size == 0) {
+    COV(PFB_X);
+    expand_buffer(sizeof(_small_buffer));
+    assert(_buffer_size > 0, "");
+    // and continue with at least a little buffer
+  }
```

Comment:
Get rid of this branch, small buffer now default.

Reply:
Fixed.